File: June2021/RemoveAllAdjacentDuplicatesInString.java
// TC : O(n)
// SC : O(n)
class Solution {
     public String removeDuplicates(String s) {
         Stack<Character> st = new Stack<>();
         for(char c : s.toCharArray()){
             if(!st.empty() && st.peek() == c){
              st.pop();
             } else {
                 st.push(c);
             }

         }
         String ans = "";
         while(!st.empty()){
             ans = st.pop() + ans;
         }
         return ans;
     }

  public String removeDuplicates(String s) {
    StringBuilder sb = new StringBuilder();
    for(char c : s.toCharArray()){
      if(sb.length()>0 && sb.charAt(sb.length()-1) == c){
        sb.deleteCharAt(sb.length()-1);
      } else{
        sb.append(c);
      }

    }
    return sb.toString();
  }
}

File: November2022/C++ Sol/remove-all-adjacent-duplicates-in-string.cpp
// @Sarang661
class Solution {
public:
    string removeDuplicates(string s) {

        stack<char>removeDup;
        int total_size = s.size();

        for(int index = 0; index < total_size; index++){
            bool flag = false;
            while(removeDup.size()>0 && removeDup.top()==s[index] ){
                flag = true;
                removeDup.pop();
            }
            if(flag==false){
                removeDup.push(s[index]);
            }

        }
        string ans = "";
        while(removeDup.empty()==false){
            ans+=removeDup.top();
            removeDup.pop();
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};